presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClientModule.java

```diff
@@ -129,6 +135,14 @@ protected JdbcClient getDelegate()
         };
     }
 
+    @Provides
+    @Singleton
+    @StatsCollecting
+    public static ConnectionFactory createConnectionFactoryWithStats(ConnectionFactory connectionFactory)
+    {
+        return new StatisticsAwareConnectionFactory(connectionFactory);
+    }
+
     @Provides
     @Singleton
     public ConnectionFactory getConnectionFactory(PhoenixConfig config)
```

Comment:
I think this one might have been marked as `@ForBaseJdbc` instead of `@TheXXX` annotations. It would seem more natural as now every user of `ConnectionFactory` has to remember that it has to be annotated with `@TheConnectionFactory` to get the right one.

Reply:
Also note that if we have used `@ForBaseJdbc` annotation, it would be easier to correctly bind stuff when writing connector. For example, when somebody binds raw `ConnectorFactory` Guice will fail to initialize because of double binding.
On the other hand, when now somebody injects raw `ConnectorFactory` he won't get any error even though he picks up the wrong one.

Reply:
This sounds good to me. @findepi What do you think?

Reply:
@electrum Do you have any thoughts about it?

Reply:
@sopel39 seems right